app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/newdevicenotice/NewDeviceNoticeEmailAccessScreen.kt

```diff
@@ -0,0 +1,194 @@
+import com.x8bit.bitwarden.ui.auth.feature.newdevicenotice.NewDeviceNoticeEmailAccessEvent.NavigateToTwoFactorOptions
+import com.x8bit.bitwarden.ui.platform.base.util.EventsEffect
+import com.x8bit.bitwarden.ui.platform.components.button.BitwardenFilledButton
+import com.x8bit.bitwarden.ui.platform.components.scaffold.BitwardenScaffold
+import com.x8bit.bitwarden.ui.platform.components.toggle.BitwardenSwitch
+import com.x8bit.bitwarden.ui.platform.components.util.rememberVectorPainter
+import com.x8bit.bitwarden.ui.platform.theme.BitwardenTheme
+
+/**
+ * The top level composable for the new device notice email access screen.
+ */
+@Composable
+fun NewDeviceNoticeEmailAccessScreen(
+    onNavigateToTwoFactorOptions: () -> Unit,
+    viewModel: NewDeviceNoticeEmailAccessViewModel = hiltViewModel(),
+) {
+    val state by viewModel.stateFlow.collectAsStateWithLifecycle()
+    EventsEffect(viewModel = viewModel) { event ->
+        when (event) {
+            NavigateToTwoFactorOptions -> onNavigateToTwoFactorOptions()
+        }
+    }
+
+    BitwardenScaffold(
+        modifier = Modifier
+            .fillMaxSize(),
+    ) {
+        NewDeviceNoticeEmailAccessContent(
+            email = state.email,
+            isEmailAccessEnabled = state.isEmailAccessEnabled,
+            onEmailAccessToggleChanged = remember(viewModel) {
+                { newState ->
+                    viewModel.trySendAction(EmailAccessToggle(newState = newState))
+                }
+            },
+            onContinueClick = { viewModel.trySendAction(ContinueClick) },
+        )
+    }
+}
+
+@Composable
+private fun NewDeviceNoticeEmailAccessContent(
+    email: String,
+    isEmailAccessEnabled: Boolean,
+    onEmailAccessToggleChanged: (Boolean) -> Unit = {},
+    onContinueClick: () -> Unit = {},
+) {
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center,
+        modifier = Modifier
+            .fillMaxSize()
+            .verticalScroll(state = rememberScrollState()),
+    ) {
+        HeaderContent()
+        Spacer(modifier = Modifier.height(24.dp))
+        MainContent(
+            email = email,
+            isEmailAccessEnabled = isEmailAccessEnabled,
+            onEmailAccessToggleChanged = onEmailAccessToggleChanged,
+        )
+        BitwardenFilledButton(
+            label = stringResource(R.string.continue_text),
+            onClick = onContinueClick,
+            modifier = Modifier
+                .padding(24.dp)
+                .fillMaxWidth()
+                .imePadding(),
+        )
+        Spacer(modifier = Modifier.navigationBarsPadding())
+    }
+}
+
+/**
+ * Header content containing the warning icon and title.
+ */
+
+@Suppress("MaxLineLength")
+@Composable
+private fun HeaderContent(
+    modifier: Modifier = Modifier,
+) {
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = modifier.padding(horizontal = 24.dp),
```

Comment:
Why is this using a non-standard amount of padding?

Can we discuss this with design because this screen is going to be difficult to maintain since it uses so many non-standard design patterns.